File: 0001-TwoSum/python/TwoSum.py
def TwoSum(nums, target):
    pair_idx = {}

    for i, num in enumerate(nums):
        if target - num in pair_idx:
            return [i, pair_idx[target - num]]
        pair_idx[num] = i 

if __name__ == "__main__":
    nums= [2,7,11,15]
    target = 9
    result = TwoSum(nums, target)
    print(result)

File: 0001-TwoSum/cpp/TwoSum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap;
        for (int i = 0; i < nums.size(); i++) {
            auto index = hashmap.find(target - nums[i]);
            if (index != hashmap.end()) {
                return {index->second, i};
            }
            hashmap.insert(pair<int, int>(nums[i], i));
        }
        return {};
    }
};

int main() {
    Solution solution;

    vector<int> nums = {2, 7, 11, 15}; 
    int target = 9;                  

    vector<int> result = solution.twoSum(nums, target);

    if (!result.empty()) {
        cout << "Indices: " << result[0] << ", " << result[1] << endl;
        cout << "Numbers: " << nums[result[0]] << ", " << nums[result[1]] << endl;
    } else {
        cout << "No solution found!" << endl;
    }

    return 0;
}
